### ISSUE TITLE
Pagination causes issues with 'page=200'

### ISSUE THREAD

**Issue by emi-x** [circleguard/ossapi#24]
When trying to pull results from page 200 (users 9951-10000) the whole thing kinda falls apart. Works for all other pages i tested including 199 but fails on 200. 

```
from ossapi import *
api = OssapiV2("xxxx", "xxxxxxxxxxxxxxxxxxxxx")
cursor = Cursor(page=200)
r = api.ranking("osu", RankingType.SCORE, cursor=cursor)
print(r.ranking[-1].user.username)
print(api.search(query=r.ranking[-1].user.username).user.data[0].id)
print(api.user(api.search(query=r.ranking[-1].user.username).user.data[0].id).statistics.ranked_score)
```

```
Traceback (most recent call last):
  File "C:/Users/xbdev/Documents/osu test.py", line 4, in <module>
    r = api.ranking("osu", RankingType.SCORE, cursor=cursor)
  File "C:\Users\xbdev\AppData\Local\Programs\Python\Python39\lib\site-packages\ossapi\ossapiv2.py", line 84, in wrapper
    return function(*args, **kwargs)
  File "C:\Users\xbdev\AppData\Local\Programs\Python\Python39\lib\site-packages\ossapi\ossapiv2.py", line 691, in ranking
    return self._get(Rankings, f"/rankings/{mode.value}/{type_.value}",
  File "C:\Users\xbdev\AppData\Local\Programs\Python\Python39\lib\site-packages\ossapi\ossapiv2.py", line 214, in _get
    return self._instantiate_type(type_, json_)
  File "C:\Users\xbdev\AppData\Local\Programs\Python\Python39\lib\site-packages\ossapi\ossapiv2.py", line 385, in _instantiate_type
    return self._resolve_annotations(value)
  File "C:\Users\xbdev\AppData\Local\Programs\Python\Python39\lib\site-packages\ossapi\ossapiv2.py", line 306, in _resolve_annotations
    value = self._instantiate_type(type_, value, obj, attr_name=attr)
  File "C:\Users\xbdev\AppData\Local\Programs\Python\Python39\lib\site-packages\ossapi\ossapiv2.py", line 380, in _instantiate_type
    value = self._instantiate(type_, value)
  File "C:\Users\xbdev\AppData\Local\Programs\Python\Python39\lib\site-packages\ossapi\ossapiv2.py", line 408, in _instantiate
    for key in list(kwargs):
TypeError: 'NoneType' object is not iterable
```

**Comment by tybug** [comment]
Fairly certain this is because there are only 200 pages of results. Let's return `None` for the cursor instead here and then it's the user's responsibility to check for a null cursor, which indicates there are no more pages.

Fixed in 261d3bf7e0c6aa5202731b5a8d022c96f6c85536. Thanks for the report

**Comment by tybug** [comment]
By the way, you don't need the intermediary `api.search` call - `api.user` accepts a username as well as an id. `print(api.user(r.ranking[-1].user.username).statistics.ranked_score)`
